src/renderers/testing/ReactShallowRenderer.js

```diff
@@ -100,26 +59,146 @@ class ReactShallowRenderer {
       element.type,
     );
 
-    // TODO Remove this wrapper if/when context parameter is removed
-    if (context && Object.keys(context).length) {
-      element = wrapElementWithContextProvider(element, context);
-    }
+    this._element = element;
 
-    this._renderer = ReactTestRenderer.create(element, {
-      createNodeMock: createShallowNodeMock(),
-    });
+    if (this._instance) {
+      this._rendered = updateClassComponent(
+        this._instance,
+        element.props,
+        context,
+      );
+    } else {
+      const prototype = element.type.prototype;
+
+      if (typeof prototype.render === 'function') {
```

Comment:
Can we keep the way we determine class vs function consistent? This seems like an ad-hoc way that doesn't match how we do this elsewhere (`prototype.isReactComponent`). I'm also not sure if this supports "factory" components (`function Foo() { return { render() { return <div /> } } }`).